### ISSUE TITLE
Building profiler on macOS M1

### ISSUE THREAD

**Issue by aclysma** [wolfpld/tracy#170]
I was able to get the profiler and client code to build on M1 and want to mention the changes I needed to make. I would PR it, but I don't know the **correct** way to fix these problems. So I'll just record the error messages and what I did to make it work:

## profiler

Grabbed v0.7.5 on tracy GitHub repo
Installed capstone, glfw, freetype via brew (ARM) - no special steps here
cd profiler/build/unix
make

```
Resolving dependencies of ../../../server/TracyWorker.cpp
clang: error: the clang compiler does not support '-march=native'
```

I removed `-march=native` from profiler/build/unix/release.mk

```
c++ -c -I/opt/homebrew/Cellar/glfw/3.3.2/include -I/opt/homebrew/opt/freetype/include/freetype2 -I/opt/homebrew/Cellar/capstone/4.0.2/include/capstone -I../../../imgui -I../../libs/gl3w -O3 -s   -std=c++17 -D"DISPLAY_SERVER_X11" -DNDEBUG -DIMGUI_IMPL_OPENGL_LOADER_GL3W ../../../server/TracySourceView.cpp -o obj/release/o/o/o/../../../server/TracySourceView.o
clang: warning: argument unused during compilation: '-s' [-Wunused-command-line-argument]
../../../server/TracySourceView.cpp:5:10: fatal error: 'capstone/capstone.h' file not found
#include <capstone/capstone.h>
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

I changed `#include <capstone/capstone.h>` to `#include <capstone.h>` in TracySourceView.cpp and TracyWorker.cpp

```
c++ -c -I/opt/homebrew/Cellar/glfw/3.3.2/include -I/opt/homebrew/opt/freetype/include/freetype2 -I/opt/homebrew/Cellar/capstone/4.0.2/include/capstone -I../../../imgui -I../../libs/gl3w -O3 -s   -std=c++17 -D"DISPLAY_SERVER_X11" -DNDEBUG -DIMGUI_IMPL_OPENGL_LOADER_GL3W ../../../server/TracyWorker.cpp -o obj/release/o/o/o/../../../server/TracyWorker.o
clang: warning: argument unused during compilation: '-s' [-Wunused-command-line-argument]
In file included from ../../../server/TracyWorker.cpp:26:
../../../server/TracyFileRead.hpp:332:23: error: variable has incomplete type 'struct stat64'
        struct stat64 buf;
                      ^
../../../server/TracyFileRead.hpp:332:16: note: forward declaration of 'stat64'
        struct stat64 buf;
               ^
1 error generated.
```

I added `#define stat64 stat`

## Client

I'm using tracy via rust bindings https://github.com/nagisa/rust_tracy_client. The only modification I made was to update the git submodule in that project to point at 0.7.5. I didn't need to make any other changes for it to compile/work. So I suspect most people consuming the client code will not need to make any code changes to compile on M1.

**Comment by wolfpld** [comment]
In "Info -> Trace statistics" what value do you get for "Timer resolution"?

**Comment by wolfpld** [comment]
All the issues have been resolved on master. (Please still answer to the question above.)

**Comment by aclysma** [comment]
41ns

**Comment by wolfpld** [comment]
Can you check if this time is improved (decreased) after applying this patch?

```diff
diff --git a/client/TracyProfiler.hpp b/client/TracyProfiler.hpp
index 530e0c8b..d880fd8e 100644
--- a/client/TracyProfiler.hpp
+++ b/client/TracyProfiler.hpp
@@ -25,7 +25,7 @@
 #  include <mach/mach_time.h>
 #endif

-#if defined _WIN32 || defined __CYGWIN__ || ( defined __i386 || defined _M_IX86 || defined __x86_64__ || defined _M_X64 ) || ( defined TARGET_OS_IOS && TARGET_OS_IOS == 1 )
+#if defined _WIN32 || defined __CYGWIN__ || ( defined __i386 || defined _M_IX86 || defined __x86_64__ || defined _M_X64 ) || ( defined TARGET_OS_IOS && TARGET_OS_IOS == 1 ) || ( defined __APPLE__ && defined __aarch64__ )
 #  define TRACY_HW_TIMER
 #endif

@@ -129,7 +129,7 @@ public:
     static tracy_force_inline int64_t GetTime()
     {
 #ifdef TRACY_HW_TIMER
-#  if defined TARGET_OS_IOS && TARGET_OS_IOS == 1
+#  if ( defined TARGET_OS_IOS && TARGET_OS_IOS == 1 ) || ( defined __APPLE__ && defined __aarch64__ )
         return mach_absolute_time();
 #  elif defined _WIN32 || defined __CYGWIN__
 #    ifdef TRACY_TIMER_QPC
```

**Comment by aclysma** [comment]
It was still 41ns with the patch. BTW I needed to change `ifeq ($(ARCH),aarch64)` to `ifeq ($(ARCH),arm64)`. Otherwise, warning-free build now, nice!

**Comment by wolfpld** [comment]
Thanks for the information.

The architecture identifier check should be fixed in e3ae2c077.
